data-integration/getHotMaps_raster.py

```diff
@@ -93,15 +93,15 @@ def get(repository: str, dp: frictionless.package.Package, isForced: bool = Fals
             }
             rasters.append(raster)
             # check statistics for each resource
-            if dp != None and "stats" in new_dp["resources"][resource_idx]:
+            if dp is not None and "stats" in new_dp["resources"][resource_idx]:
                 if (
                     dp["resources"][resource_idx]["stats"]
                     != new_dp["resources"][resource_idx]["stats"]
                 ):
                     isChangedStats = True
     rasters = pd.DataFrame(rasters)
 
-    if dp != None:  # Existing dataset
+    if dp is not None:  # Existing dataset
```

Comment:
writing

```python
if dp: 
```
is concidered more pythonic.

Reply:
dp is never a boolean, I don't think it would work in this case

Reply:
Yes it works, because you are testing for identity with "None", not if dp has the value "True" or "False".
"True" is not "None", but "False" is not "None" either.

Reply:
Not in this case:
```
dp = {}
if dp:
	print("test1")


if dp is not None:
	print("test2")
```
Anyway, I don't think this is the point of this commit. And the non-pythonic way was accepted in the previous merge. Here the change was from `!=` to `is not`, because the previous version failed the precommit.

Reply:
No, this is wrong and you should change it.

Your example doesn't print "test1", because if you test an empty dictionary, with no keys, it returns "False".
```
>>> dp1 = {}
>>> dp2 = {"version": None}
>>>
>>> if dp1 is not None: print("test1")
... 
prints test1
>>> if dp1: print("test2")
...
prints nothing
>>> if dp2 is not None: print("test3")
...
prints test3
>>> if dp2: print("test4")
...
prints test4
>>>
```

In the following lines of your code, you make the test `isChangedVersion = dp["version"] != new_dp["version"]` which requires that you have the key "version" existing in your dictionary.

But if you check your dictionary like that `if dp is not None`, you may encounter a key error:
```
>>> dp1["version"] == dp2["version"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'version'
```

that can't happen if you check your dictionary with `if dp`.

Reply:
Ok, I will correct it. Can you please open an issue for that?

Reply:
Here maybe some piece of explanation for all: `https://www.python.org/dev/peps/pep-0008/` under `programming recommendations`

```
Comparisons to singletons like None should always be done with `is` or `is not`, never the equality operators.

Also, beware of writing `if x` when you really mean `if x is not None` 
-- e.g. when testing whether a variable or argument that defaults to `None` was set to some other value. 
The other value might have a type (such as a container) that could be false in a boolean context!
```

Reply:
> 
> 
> Ok, I will correct it. Can you please open an issue for that?

Why open an issue for that ?